### PR TITLE
Update proxy.py to set a timeout for incoming socket connections.

### PR DESCRIPTION
**Pull request by ojarjur**
This change sets a timeout for accepted incoming socket connections in the tunnel-over-websocket proxy.

Without this, sockets inherit the default timeout of `None`, which causes them to be blocking but never time out.

### CODE REVIEW

File: google/cloud/dataproc_spark_connect/client/proxy.py [hunk 1]
```diff
@@ -192,6 +192,7 @@ def _run(self, s):
             s.release()
             while not self._killed:
                 conn, addr = frontend_socket.accept()
+                conn.settimeout(1)
```
Comment (medb) [inline google/cloud/dataproc_spark_connect/client/proxy.py:195]: Isn't this too low for conn timeout? Maybe 3-5s for safety?

Comment (ojarjur) [reply google/cloud/dataproc_spark_connect/client/proxy.py:213]: The safety of the code isn't based on the length of the timeout, but rather on how gracefully it handles timeout errors.

However, your comment made me go back over the code to make sure we handled every scenario correctly, and in doing that I found one scenario where we did not.

I've added a unit test that reproduces that edge case (along with many others), and then fixed the code and used this test case to verify the fix is correct.

Comment (medb) [reply google/cloud/dataproc_spark_connect/client/proxy.py:213]: Thanks! I think it may be good to clarify this in the comment to this line - I thought that this may impact whether connection could be established, etc if there is slowness in the networks, but it seems like this is not the case.

**Review by medb — APPROVED** [review]

File: google/cloud/dataproc_spark_connect/client/proxy.py (hunk 1 above)
Comment (ojarjur) [reply google/cloud/dataproc_spark_connect/client/proxy.py:220]: Done.
